ffi/vmlib/src/lib.rs

```diff
@@ -32,29 +32,28 @@ impl AthconVm for AthenaVMWrapper {
   }
 
   /// `execute` is the main entrypoint from FFI. It's called from the macro-generated `__athcon_execute` fn.
-  /// Note that we have to pass in raw `host` and `context` pointers here. If we wrap them into the
+  /// Note that we have to pass in raw `context` pointer here. If we wrap it into the
   /// `AthenaExecutionContext` object inside the top-level FFI function and pass it in here, it causes
   /// lifetime issues.
-  unsafe fn execute<'a>(
+  fn execute(
     &self,
     rev: Revision,
     code: &[u8],
     message: &AthconExecutionMessage,
-    host: *const ffi::athcon_host_interface,
+    host: &ffi::athcon_host_interface,
     context: *mut ffi::athcon_host_context,
   ) -> AthconExecutionResult {
     // note that host context is allowed to be null. it's opaque and totally up to the host
-    // whether and how to use it. but we require the host interface to be non-null.
-    if host.is_null() || message.kind() != AthconMessageKind::ATHCON_CALL || code.is_empty() {
+    // whether and how to use it.
+    if message.kind() != AthconMessageKind::ATHCON_CALL || code.is_empty() {
       return AthconExecutionResult::failure();
     }
 
     // Perform the conversion from `ffi::athcon_message` to `AthenaMessage`
     let athena_msg = AthenaMessageWrapper::from(message);
 
     // Unpack the context
-    let host_interface: &ffi::athcon_host_interface = unsafe { &*host };
-    let execution_context = AthconExecutionContext::new(host_interface, context);
+    let execution_context = ExecutionContext::new(host, context);
```

Comment:
What happens here if `host` is null?

Reply:
It cannot be `null` here, it's guaranteed because it's a reference, not a pointer. The caller must have a valid reference to an `athcon_host_interface` to call `execute()`.